tools/local-snapshot-runner-app/src/exportFile.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Loader } from "@fluidframework/container-loader";
+import { createLocalOdspDocumentServiceFactory } from "@fluidframework/odsp-driver";
+import { ChildLogger, PerformanceEvent } from "@fluidframework/telemetry-utils";
+import * as fs from "fs";
+import FileLogger from "./logger/FileLogger";
+import { getArgsValidationError } from "./getArgsValidationError";
+import { isCodeLoaderBundle } from "./codeLoaderBundle";
+import { FakeUrlResolver } from "./fakeUrlResolver";
+import path from "path";
+import { ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+
+export async function exportFile(
+    codeLoader: string,
+    inputFile: string,
+    outputFolder: string,
+    scenario: string,
+    telemetryFile: string,
+    loggerOverride?: ITelemetryBaseLogger,
+) {
+    if (fs.existsSync(telemetryFile)) {
+        console.error("Telemetry file already exists. " + telemetryFile);
+        throw new Error("Telemetry file already exists.");
```

Comment:
I'd suggest having a delegate that does lines 26-27.
- For stand-alone tool, it should print to console and invoke process.exit(1) - users do not need to see long stack for where issue happened.
    - That said, even for stand-alone tool we may choose to use it (morph it to take different form) to process many files in a row - in such case process.exit() is not the right tool.
- But if we leverage the "core" of the solution somewhere else (see my other comment), I'd want to throw exception and not print anything to console.
- if I host it in multi-threaded node.js process, I may want to communicate error to main thread and exit thread (see how it's done in snapshot tests).

Basically, I'd love to see code reusable, and only very small piece of code assume that it's stand-alone node.js executable.

Reply:
Or maybe even better - not do it through delegate, but through strongly typed return type - it's either success or error of some sort. And outer code controls how this error is handled.
We better not rely on exceptions for that - we regret that same choice in drivers - exceptions should be used for exceptional cases (like logical errors in our logic, out of memory, etc.). Everything else - structured (typed) return types.

In this case (as opposed to a driver example) one type of error is likely sufficient.